### ISSUE TITLE
Phase09-Part1-Team3

### ISSUE THREAD

**Issue by kimianoorbakhsh** [Star-Academy/codestar-intern-issues#87]
This issue template is based of [Phase09-Web.md](https://github.com/Star-Academy/codestar-internship/blob/master/Projects/Phase09-Web/Phase09-Web.md) contents.
Click on "Raw" button on right side of this page to see original markdown then create a new issue with orginal content of this tasklist:

# Phase 9 First Part

- [x] Before starting: Read the link about history and purpose of creating & developing Web, then answer these questions:
  - [x] How did developers provide server-client software for users before the creation of web technologies?
  
      ` Every app had its pre-compiled client program and it had to be separately installed on every user's PC. `
      
  - [x] What was the main trouble with traditional technologies before the web?
  
      `The components of client and server were tightly bound to the definite operating system and computer architecture. As a 
        result, it was expensive to port apps to other systems.`
      
  
  - [x] How did the web overcome this main trouble?
  
      ` A web application is a client-software application run by the client in a browser. The main function of a browser is to show the information received from a server and send the user's data back. The main advantage of this approach is the fact that clients do not depend on the user's specific operating system; therefore, web applications are cross-platform services.`

## Asp.net core

- [x] Getting Started with HTTP:
  - [x] What does a protocol mean?
  
    `A protocol is a system of rules that define how data is exchanged within or between computers.`

  - [x] Why designing a protocol for communication is an important requirement of all other web-related technologies?
  
    `Because communications between devices require that the devices agree on the format of the data that is being exchanged. The set of rules that defines a format is called a protocol.`
    
  - [x] Try to get familiar with HTTP by reading links.
  - [x] Why HTTP is the most popular protocol in web technologies?
    
    ` HTTP is an extensible protocol that is easy to use. The client-server structure, combined with the ability to simply add headers, allows HTTP to advance along with the extended capabilities of the Web and also it is used to not only fetch hypertext documents but also images and videos or to post content to servers, like with HTML form results. HTTP can also be used to fetch parts of documents to update Web pages on demand.`
    
  - [x] What is the most interesting aspect of HTTP in your opinion? 
  
    `HTTP is generally designed to be simple and human-readable`

- [x] Getting Started with REST:
  - [x] Read mentioned links.
  - [x] Why do we need RESTful APIs?
  
    `By using Rest you have a service that a client developer can be able to easily understand due to the consistent use of the HTTP protocol. The service will be easily discoverable due to 
hypermedia and the client will be extremely resilient to changes on the server.`

- [x] Getting Started with Asp.net Core:
  - [x] Read mentioned links.
  - [x] Create a hello world API.
  - [x] Call your api from browser.
  - [x] Play with routing options! (Try to use it & enjoy it!)
  - [x] Why do we need configurable routings in web development?
  
    `To match the URLs of incoming requests and map them to actions.  Placing a route on the controller or action makes it attribute-routed`
  
  - [x] Play with dependency injection! (Define an interface & inject interface implementation to code)
  - [x] Why do we need dependency injection?
  
    `Without dependency injection , implementation is difficult to unit test and configuration code becomes scattered across the app. `
  - [x] Play with parameter binding! (use different modes)
  - [x] Install postman.
  - [x] Call your APIs with postman.
  
- [x] Implement Asp.net Core Project:
  - [x] Design your search API.
  - [x] Implement & Test your API.

# Review
Link of your own PR:

`[FILL HERE WITH LINK OF YOUR PR.]`

Link of other PR(s) you agreed to review:

[Team-2](https://github.com/Star-Academy/Team2-Codes/pull/7
)

- [x] Your own PR is reviewed and approved by at least one other team.

- [x] Your own PR is reviewed and approved by mentor (hamyar).

- [x] Your own PR is reviewed and approved by manager (rahbar).

- [x] Your own PR is merged.
